### DISCUSSION TITLE
constraint does not work

### DISCUSSION THREAD

**Discussion by faranak1991** [anyoptimization/pymoo#599] [General]
Hello,

I am using the following code to ensure that the offspring x data set satisfies this condition (by define a constraint) :
abs.(3 * x[0] + 1* x[1] + 1* x[2] + 3* x[3] + 3* x[4] + 3* x[5] + 3* x[6] + 6* x[7] + 6* x[8] + 6* x[9] + 9* x[10] ) < 2e-2 

However, after tracking the output, I noticed that the individual's parameters x are not satisfying this condition. 

Any help to identify and resolve the issue would be greatly appreciated ?

class myProblem(Problem):
    def __init__(self, n_var, lower_bounds, upper_bounds):
        super().__init__(n_var=n_var, n_obj=5, n_ieq_constr=1, xl=lower_bounds, xu=upper_bounds)

    def _evaluate(self, X, out, *args, **kwargs):
        n_samples = X.shape[0]
        f1 = np.empty((n_samples, 1))
        f2 = np.empty((n_samples, 1))
        f3 = np.empty((n_samples, 1))
        f4 = np.empty((n_samples, 1))
        f5 = np.empty((n_samples, 1))
        g = np.empty((n_samples, 1))  

        for i in range(n_samples):
            x = X[i, :]  
            objectives = objective_function(x)  

            f1[i, 0] = objectives[0]  
            f2[i, 0] = objectives[1]   
            f3[i, 0] = objectives[2]   
            f4[i, 0] = objectives[3]   
            f5[i, 0] = objectives[4]   

            factors = np.array([3, 1, 1, 3, 3, 3, 3, 6, 6, 6, 9])
            constraint_value = np.abs(np.dot(factors, x))
            constraint_violation = constraint_value - 2e-2
            g[i, 0] = constraint_violation

        out["F"] = np.column_stack([f1, f2, f3, f4, f5])
        out["G"] = g

**Comment by blankjul** [comment]
I am not sure about your objective function but in fact the constraint violation can be satisfied.

```
import numpy as np

from pymoo.algorithms.soo.nonconvex.ga import GA
from pymoo.core.problem import Problem
from pymoo.optimize import minimize


class MyProblem(Problem):

    def __init__(self):
        super().__init__(n_var=11, n_obj=1, n_ieq_constr=1, xl=0, xu=1)

    def _evaluate(self, X, out, *args, **kwargs):
        n = X.shape[0]
        f = np.empty((n, 1))
        g = np.empty((n, 1))

        for i, x in enumerate(X):
            x = X[i, :]
            f[i, 0] = 0

            factors = np.array([3, 1, 1, 3, 3, 3, 3, 6, 6, 6, 9])
            constraint_value = np.abs(np.dot(factors, x))
            constraint_violation = constraint_value - 2e-2
            g[i, 0] = constraint_violation

        out["F"] = f
        out["G"] = g


problem = MyProblem()

algorithm = GA(pop_size=100, eliminate_duplicates=True)

res = minimize(problem,
               algorithm,
               seed=1,
               verbose=True)

print("Best solution found: \nX = %s\nF = %s\nCV = %s" % (res.X, res.F, res.CV))

```

**Comment by faranak1991** [comment]
Thanks for your reply, it works.
